internal/ingress/controller/nginx.go

```diff
@@ -384,6 +384,11 @@ func (n *NGINXController) Stop() error {
 	n.stopLock.Lock()
 	defer n.stopLock.Unlock()
 
+	if n.cfg.WaitBeforeShutdown > 0 {
```

Comment:
Please check my comment https://github.com/kubernetes/ingress-nginx/pull/4006#issuecomment-547913361
This should be before we stop nginx

Reply:
> @miklosn because of the @ElvinEfendi comment and the impact of the change, please add e2e tests.
> You can use https://github.com/kubernetes/ingress-nginx/blob/master/test/e2e/gracefulshutdown/slow_requests.go as a guide

Thanks guys for the comments! will get back